frontend/dockerfile/dockerfile_provenance_nydus_test.go

```diff
@@ -0,0 +1,103 @@
+//go:build nydus
+// +build nydus
+
+package dockerfile
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/containerd/containerd/platforms"
+	"github.com/containerd/continuity/fs/fstest"
+	"github.com/containerd/nydus-snapshotter/pkg/converter"
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	"github.com/moby/buildkit/client"
+	"github.com/moby/buildkit/frontend/dockerui"
+	"github.com/moby/buildkit/identity"
+	"github.com/moby/buildkit/solver/llbsolver/provenance"
+	"github.com/moby/buildkit/util/contentutil"
+	"github.com/moby/buildkit/util/testutil"
+	"github.com/moby/buildkit/util/testutil/integration"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+func init() {
+	allTests = append(allTests, integration.TestFuncs(
+		testProvenanceForNydus,
+	)...)
+}
+
+func testProvenanceForNydus(t *testing.T, sb integration.Sandbox) {
```

Comment:
Any reason this shouldn't be a `client_nydus_test`? I don't think this behavior is specific to just the `dockerfile` frontend.

Reply:
Looks like the `llbsolver.provenance` package has `AddBuildConfig` which creates the dependency.

I *think* we should probably move `AddBuildConfig` up a level into `provenance.go` - see https://github.com/moby/buildkit/pull/3926.

Reply:
@jedevc Thanks for the work, updated, PTAL.